File: lab/examples/go/internal/hexagon/application/order/cancelorder/ports/outbound/notifier.go
// Package outbound holds the contracts the cancel-order slice needs from the
// outside world.
//
// Notifier is a use-case-local port: announcing a cancellation is a need
// specific to this slice, so the port lives inside it.
package outbound

import (
	"context"

	"hexslice/example/internal/hexagon/domain/order"
)

// Notifier announces that an order has been cancelled.
type Notifier interface {
	OrderCancelled(ctx context.Context, id order.ID, reason string) error
}

File: lab/examples/kotlin/hexagon/application/src/main/kotlin/hexslice/hexagon/application/order/cancelorder/ports/outbound/Notifier.kt
// The outbound (driven) port that only the cancel-order slice needs.
//
// Announcing a cancellation is a need specific to this slice, so the port lives
// inside it.
package hexslice.hexagon.application.order.cancelorder.ports.outbound

import hexslice.hexagon.domain.order.OrderId

/** Announces that an order has been cancelled. */
interface Notifier {
    fun orderCancelled(id: OrderId, reason: String)
}
